src/utils/utils.ts

```diff
@@ -10,6 +30,39 @@ export const fetchSameKeyDocs = async (
   return await Promise.allSettled(docPromisesArr)
 }
 
+export const fetchExchangeRates = async (
+  wallets: EdgeCurrencyWallet[]
+): Promise<ExchangeRateInfo> => {
+  const ratePromisesArr = wallets.map(async wallet => {
+    const {
+      currencyInfo: { currencyCode }
+    } = wallet
+    try {
+      const response = await fetch(baseUri + route + queryStr + currencyCode)
+      const { exchangeRate } = asRatesServerResponse(await response.json())
+      const exchangeRateFloat = parseFloat(exchangeRate)
+
+      // Check if the response was successful or if the exchange rate is a valid number
+      if (!response.ok || isNaN(exchangeRateFloat)) {
+        throw new TypeError(fetchErrorMsg + currencyCode)
+      }
+
+      return { [currencyCode]: exchangeRateFloat }
+    } catch (e) {
+      console.log(e)
+    }
+  })
+
+  // Resolve array of promises for exchange rates
+  const rates = await Promise.all(ratePromisesArr)
+  const exchangeRatesObj = {}
+  for (const rate of rates) {
+    if (rate === undefined) continue
```

Comment:
What situation would lead to rate being undefined?

Reply:
The promise will return `undefined` when an error is thrown because the fetch for the exchange rate was not a successful response (200-299), or the data returned was not a number (lines 46 - 48). After the error is thrown, it will be caught at lines 51-52 where the error will be logged. Since nothing is explicitly returned, `undefined` will be returned by default.